### PR TITLE
executor: replace logger with zap logger

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

Unify log format in package `executor`.

### What is changed and how it works?

Replace log package with zap logger and follow the new log format.

### Check List 

Tests 

 - No code

Code changes

 - None

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/admin.go [hunk 1]
```diff
@@ -393,8 +395,9 @@ func (e *RecoverIndexExec) batchMarkDup(txn kv.Transaction, rows []recoverRows)
 				}
 
 				if handle != rows[i].handle {
-					log.Warnf("[recover-index] The constraint of unique index:%v is broken, handle:%v is not equal handle:%v with idxKey:%v.",
-						e.index.Meta().Name.O, handle, rows[i].handle, key)
+					log.Warn("The constraint of unique index is broken when recovering index",
+						zap.String("index name", e.index.Meta().Name.O), zap.ByteString("key value", key),
```
Comment (qw4990) [inline executor/admin.go]: Using `zap.ByteString("key", key)` is better?

Comment (qw4990) [reply executor/admin.go]: And should we keep the key as a single word obeying Go style like `indexName` instead of `index name`?

Comment (lamxTyler) [reply executor/admin.go]: Yes, you are right.

File: executor/aggregate.go [hunk 2]
```diff
@@ -318,9 +319,10 @@ func (w *HashAggPartialWorker) getChildInput() bool {
 }
 
 func recoveryHashAgg(output chan *AfFinalResult, r interface{}) {
-	output <- &AfFinalResult{err: errors.Errorf("%v", r)}
+	err := errors.Errorf("%v", r)
+	output <- &AfFinalResult{err: err}
 	buf := util.GetStack()
-	log.Errorf("panic in the recoverable goroutine: %v, stack trace:\n%s", r, buf)
+	log.Error("Panic in the recoverable goroutine", zap.Error(err), zap.String("stack trace", string(buf)))
```
Comment (crazycs520) [inline executor/aggregate.go]: /s/"stack trace"/"stack" ?

File: executor/ddl.go [hunk 3]
```diff
@@ -252,7 +253,7 @@ func (e *DDLExec) executeDropTableOrView(s *ast.DropTableStmt) error {
 		}
 
 		if config.CheckTableBeforeDrop {
-			log.Warnf("admin check table `%s`.`%s` before drop.", fullti.Schema.O, fullti.Name.O)
+			logutil.Logger(context.Background()).Warn("admin check table before drop", zap.String("tableName", fmt.Sprintf("`%s`.`%s`", fullti.Schema.O, fullti.Name.O)))
```
Comment (zz-jason) [inline executor/ddl.go]: how about:
```go
logutil.Logger(context.Background()).Warn("admin check table before drop",
	zap.String("database", fullti.Schema.O),
	zap.String("table", fullti.Name.O),
)
```

File: executor/admin.go [hunk 4]
```diff
@@ -393,8 +395,9 @@ func (e *RecoverIndexExec) batchMarkDup(txn kv.Transaction, rows []recoverRows)
 				}
 
 				if handle != rows[i].handle {
-					log.Warnf("[recover-index] The constraint of unique index:%v is broken, handle:%v is not equal handle:%v with idxKey:%v.",
-						e.index.Meta().Name.O, handle, rows[i].handle, key)
+					logutil.Logger(context.Background()).Warn("recover index: the constraint of unique index is broken, handle in index is not equal to handle in table",
+						zap.String("indexName", e.index.Meta().Name.O), zap.ByteString("indexKey", key),
```
Comment (XuHuaiyu) [inline executor/admin.go]: Why use `indexName` here but `index` in line 313

File: executor/aggregate.go [hunk 5]
```diff
@@ -317,9 +317,9 @@ func (w *HashAggPartialWorker) getChildInput() bool {
 }
 
 func recoveryHashAgg(output chan *AfFinalResult, r interface{}) {
+	err := errors.Errorf("%v", r)
 	output <- &AfFinalResult{err: errors.Errorf("%v", r)}
-	buf := util.GetStack()
-	log.Errorf("panic in the recoverable goroutine: %v, stack trace:\n%s", r, buf)
+	logutil.Logger(context.Background()).Error("panic in the recoverable goroutine", zap.Error(err))
```
Comment (XuHuaiyu) [inline executor/aggregate.go]: Will `parallel hash aggregation panicked` be better?

File: executor/distsql.go [hunk 6]
```diff
@@ -713,7 +714,7 @@ func (w *tableWorker) pickAndExecTask(ctx context.Context) {
 			buf := make([]byte, 4096)
 			stackSize := runtime.Stack(buf, false)
 			buf = buf[:stackSize]
-			log.Errorf("tableWorker panic stack is:\n%s", buf)
+			logutil.Logger(ctx).Error("table worker panicked", zap.String("stack", string(buf)))
```
Comment (XuHuaiyu) [inline executor/distsql.go]: tableWorker in IndexLookUpExecutor panicked?

File: executor/compiler.go [hunk 7]
```diff
@@ -76,7 +77,7 @@ func logExpensiveQuery(stmtNode ast.StmtNode, finalPlan plannercore.Plan) (expen
 	if len(sql) > logSQLLen {
 		sql = fmt.Sprintf("%s len(%d)", sql[:logSQLLen], len(sql))
 	}
-	log.Warnf("[EXPENSIVE_QUERY] %s", sql)
+	logutil.Logger(context.Background()).Warn("[EXPENSIVE_QUERY]", zap.String("sql", sql))
```
Comment (XuHuaiyu) [inline executor/compiler.go]: `[`,`]` are useless
s/sql/SQL

File: executor/distsql.go [hunk 8]
```diff
@@ -612,7 +613,7 @@ func (w *indexWorker) fetchHandles(ctx context.Context, result distsql.SelectRes
 			buf := make([]byte, 4096)
 			stackSize := runtime.Stack(buf, false)
 			buf = buf[:stackSize]
-			log.Errorf("indexWorker panic stack is:\n%s", buf)
+			logutil.Logger(ctx).Error("index worker panicked", zap.String("stack", string(buf)))
```
Comment (XuHuaiyu) [inline executor/distsql.go]: indexWorker in IndexLookupExecutor panicked ?

File: executor/index_lookup_join.go [hunk 9]
```diff
@@ -305,7 +306,7 @@ func (ow *outerWorker) run(ctx context.Context, wg *sync.WaitGroup) {
 			buf := make([]byte, 4096)
 			stackSize := runtime.Stack(buf, false)
 			buf = buf[:stackSize]
-			log.Errorf("outerWorker panic stack is:\n%s", buf)
+			logutil.Logger(ctx).Error("outerWorker panicked", zap.String("stack", string(buf)))
```
Comment (XuHuaiyu) [inline executor/index_lookup_join.go]: zap.ByteString ?

Comment (lamxTyler) [reply executor/index_lookup_join.go]: Then it is harder to read the message?

File: executor/admin.go [hunk 10]
```diff
@@ -520,8 +523,8 @@ func (e *CleanupIndexExec) deleteDanglingIdx(txn kv.Transaction, values map[stri
 				}
 				e.removeCnt++
 				if e.removeCnt%e.batchSize == 0 {
-					log.Infof("[cleaning up dangling index] table: %v, index: %v, count: %v.",
-						e.table.Meta().Name.String(), e.index.Meta().Name.String(), e.removeCnt)
+					logutil.Logger(context.Background()).Info("cleaning up dangling index", zap.String("table", e.table.Meta().Name.String()),
```
Comment (eurekaka) [inline executor/admin.go]:
```suggestion
					logutil.Logger(context.Background()).Info("clean up dangling index", zap.String("table", e.table.Meta().Name.String()),
```

File: executor/aggregate.go [hunk 11]
```diff
@@ -473,7 +473,7 @@ func (w *HashAggFinalWorker) getFinalResult(sctx sessionctx.Context) {
 		partialResults := w.getPartialResult(sctx.GetSessionVars().StmtCtx, []byte(groupKey), w.partialResultMap)
 		for i, af := range w.aggFuncs {
 			if err := af.AppendFinalResult2Chunk(sctx, partialResults[i], result); err != nil {
-				log.Error(errors.ErrorStack(err))
+				logutil.Logger(context.Background()).Error("HashAggFinalWorker append final result to Chunk failed", zap.Error(err))
```
Comment (eurekaka) [inline executor/aggregate.go]:
```suggestion
				logutil.Logger(context.Background()).Error("HashAggFinalWorker failed to append final result to Chunk", zap.Error(err))
```

File: executor/load_data.go [hunk 12]
```diff
@@ -247,8 +248,8 @@ func (e *LoadDataInfo) InsertData(prevData, curData []byte) ([]byte, bool, error
 		e.rowCount++
 		if e.maxRowsInBatch != 0 && e.rowCount%e.maxRowsInBatch == 0 {
 			reachLimit = true
-			log.Infof("This insert rows has reached the batch %d, current total rows %d",
-				e.maxRowsInBatch, e.rowCount)
+			logutil.Logger(context.Background()).Info("this insert rows has reached the batch limit", zap.Int("maxBatchRows", e.maxChunkSize),
```
Comment (eurekaka) [inline executor/load_data.go]:
```suggestion
			logutil.Logger(context.Background()).Info("batch limit hit when inserting rows", zap.Int("maxBatchRows", e.maxChunkSize),
```

File: executor/admin.go [hunk 13]
```diff
@@ -393,8 +395,9 @@ func (e *RecoverIndexExec) batchMarkDup(txn kv.Transaction, rows []recoverRows)
 				}
 
 				if handle != rows[i].handle {
-					log.Warnf("[recover-index] The constraint of unique index:%v is broken, handle:%v is not equal handle:%v with idxKey:%v.",
-						e.index.Meta().Name.O, handle, rows[i].handle, key)
+					logutil.Logger(context.Background()).Warn("recover index: the constraint of unique index is broken, handle in index is not equal to handle in table",
```
[the remark this answers is not in the document]
Comment (lamxTyler) [reply executor/admin.go]: Why? I think it is better now.

File: executor/adapter.go [hunk 14]
```diff
@@ -390,13 +390,11 @@ func (a *ExecStmt) LogSlowQuery(txnTS uint64, succ bool) {
 	}
 	execDetail := sessVars.StmtCtx.GetExecDetails()
 	if costTime < threshold {
-		if logutil.SlowQueryLogger.IsLevelEnabled(log.DebugLevel) {
```
Comment (lysu) [inline executor/adapter.go]: I think this is useful :rofl:

Comment (lamxTyler) [reply executor/adapter.go]: It is already guaranteed by https://github.com/pingcap/tidb/pull/9521/files#diff-350127760839dbfd52d23927f7ff2d95R374.

**Pull request merged**